pyat/at/physics/dynamic_aperture.py

```diff
@@ -0,0 +1,895 @@
+    def dict_def_range(self, new_value):
+        self._dict_def_range = new_value
+        self.test_points_grid()  # recompute grid points if ranges are changed
+    """
+
+    def __init__(self,
+                 ring,
+                 dpp=0.0,
+                 mode=None,
+                 start_index=0,
+                 grid_mode='grid',
+                 compute_range=False,
+                 n_turns=2**10,
+                 search_divider=10,
+                 use_radiation=True
+                 ):
+        """
+        Acceptance6D computes the 6D acceptance for a pyAT lattice
+
+        :param ring: pyAT lattice
+        :param dpp: momentum deviation
+        :param mode: mode for computation. None = 6D (default), 'x-y', 'delta-x'. 'x-xp',...
+        :param start_index: index in ring where to start the computation
+        :param grid_mode: 'grid' or 'radial' set of test points to compute Acceptance
+        :param compute_range: compute range for each plane with more than 1 point to scan
+        :param nturns: number of turns that a particle must survive
+        :param search_divider: division of range used by recursive range
+        :param use_radiation: default true (turn on radiation, in future version, use radiation state of input lattice)
+        : n_point: dictionary to determine the # points to scan in each dimension
+        : dict_def_range: default range for each dimension
+        : dict_unit:  default units for each dimension
+
+        """
+
+        self.parallel_computation = True
+
+        # rotate lattice to given element
+        self.ring = at.Lattice(ring[start_index:] + ring[:start_index])
+
+        self.dpp = dpp
+        self.mode = mode
+
+        use_radiation = ring.radiation
+
+        self.ring.radiation_off()
+
+        # if self.dpp != 0.0:
+        self.orbit = self.ring.find_orbit4(dp=self.dpp)  # dpp is added to orbit here
+        #else:
+        #    self.orbit = self.ring.find_orbit6()
+
+        if use_radiation:
+            self.ring.radiation_on()
+        else:
+            self.ring.radiation_off()
+
+        self.coordinates = {'x': [],
+                            'xp': [],
+                            'y': [],
+                            'yp': [],
+                            'delta': [],
+                            'ct': []
+                            }
+        self.survived = []
+
+        self.number_of_turns = n_turns
+        self.search_divider = search_divider
+
+        self.grid_modes = ['grid', 'radial']
+        self.grid_mode = self.grid_modes[0]
+
+        # point to scan in each dimension
+        if mode == 'x-y':
+            self.n_points = {'x': 13,
+                             'xp': 1,
+                             'y': 13,
+                             'yp': 1,
+                             'delta': 1,
+                             'ct': 1
+                             }
+        elif mode == 'delta-x':
+            self.n_points = {'x': 13,
```

Comment:
This could be achieved much easier with a dictionary:
in the class variables:
```python
    default_n_points = {'x': 13, 'xp': 1, 'y': 13, 'yp': 1, 'delta': 1, 'ct': 1 }
    mode_n_points = {
        'x-y':      {'x': 13, 'xp': 1, 'y': 13, 'yp': 1, 'delta': 1, 'ct': 1 },
        'delta-x':  {'x': 13, 'xp': 1, 'y': 1, 'yp': 1, 'delta': 13, 'ct': 1},
 #       '6D':       {'x': 13, 'xp': 1, 'y': 13, 'yp': 1, 'delta': 1, 'ct': 1 },
        'xp-yp':    {'x': 1, 'xp': 13, 'y': 1, 'yp': 13, 'delta': 1, 'ct': 1},
        'x-xp':     {'x': 13, 'xp': 13, 'y': 1, 'yp': 1, 'delta': 1, 'ct': 1},
        'y-yp':     {'x': 1, 'xp': 1, 'y': 13, 'yp': 13, 'delta': 1, 'ct': 1},
        'ct-delta': {'x': 1, 'xp': 1, 'y': 1, 'yp': 1, 'delta': 13, 'ct': 13},
    }
```
and in the constructor, a single line:
```python
self.n_points = mode_n_points.get(mode, default_n_points)
```

Reply:
The class code at initialization is cumbersome, I agree. 
For the user it seemed simpler to define a mode (self.mode = 'x-y') and then define a number of points to scan setting self.n_points) . In fact there is presently no check that for a given mode the number of points are correctly corresponding. The user is in fact free to set `self.mode = 'x-y'` and `self.n_points['x'] = 1 ` and that will fail. 
I will think more about this

Reply:
I have nothing against the choice you made for inputs, my proposal is just a cleaner way of doing exactly the same thing !